api-registration/server/index.js

```diff
@@ -0,0 +1,66 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({
+  // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  const pass = argon2.hash(password);
+
+  pass.then(result => {
+    const hashedPassword = result;
+    const params = [username, hashedPassword];
+    const sql = `insert into
+  "users" ("username", "hashedPassword")
+  values($1, $2)
+  returning "userId", "username", "createdAt";`;
+    db.query(sql, params)
+      .then(result => {
+        const newUser = result.rows;
+
+        res.status(201).json({ newUser });
```

Comment:
Here the data sent back to the client should be an object with the user data directly in it. As is it creates an object with a `newUser` property that is set to an array with the user object inside of that. That makes it tough on the front end developers, the server should format the data in the most useful way possible.
```JAVASCRIPT
const newUser = result.rows[0];

res.status(201).json(newUser);
```